examples/api/lib/widgets/sliver/sliver_coordinator.0.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+void main() {
+  runApp(const SliverCoordinatorExampleApp());
+}
+
+class SliverCoordinatorExampleApp extends StatelessWidget {
+  const SliverCoordinatorExampleApp({ super.key });
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(home: SliverCoordinatorExample());
+  }
+}
+
+class SliverCoordinatorExample extends StatefulWidget {
+  const SliverCoordinatorExample({ super.key });
+
+  @override
+  State<SliverCoordinatorExample> createState() => _SliverCoordinatorExampleState();
+}
+
+class _SliverCoordinatorExampleState extends State<SliverCoordinatorExample> {
+  static const String alignedItemId = 'alignedItem';
+  late final ScrollController scrollController;
+
+  @override
+  void initState() {
+    super.initState();
+    scrollController = ScrollController();
+  }
+
+  @override
+  void dispose() {
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  void autoScrollTo(double offset) {
+    scrollController.position.animateTo(
+      offset,
+      duration: const Duration(milliseconds: 300),
+      curve: Curves.easeInOut,
+    );
+  }
+
+  // Called each time a scroll gesture ends. If the alignedItem overlaps
+  // either end of the CustomScrollView's viewport we'll auto-scroll
+  // so that it's aligned with the top or bottom.
+  void maybeAutoScrollAlignedItem(SliverCoordinatorData data) {
+    final SliverConstraints constraints = data.getSliverConstraints(alignedItemId);
+    final SliverGeometry geometry = data.getSliverGeometry(alignedItemId);
```

Comment:
This really requires folks to learn slivers deeply, are there common scenarios we might like to provide methods for instead as far as querying the state of a given coordinated sliver?

Reply:
Maybe we should at least have an isVisible getter, to quickly skip doing work if it is not even visible at all.

Reply:
If we attached this info to the ScrollableState instead of a SliverCoodinator, then the SliverCoordinatorData object could be dispensed with and something like `ScrollableState.of(context).hasSliverLayoutInfo(key)` would obviate the public SliverCoordinatorData class. More on this below.

Reply:
I don't think ScrollableState has any knowledge of what is visible. Slivers are managed by the Viewport.